Source/DafnyCore/Triggers/ComprehensionTriggerGenerator.cs

```diff
@@ -230,5 +230,22 @@ internal void CommitTriggers(SystemModuleManager systemModuleManager) {
         triggerWriter.CommitTrigger(reporter, partWriters.Count > 1 ? index : null, systemModuleManager);
       }
     }
+
+    public List<List<Expression>> GetTriggers(bool includeTriggersThatRequireNamedExpressions) {
+      var triggers = new List<List<Expression>>();
+      foreach (var triggerWriter in partWriters) {
+        if (includeTriggersThatRequireNamedExpressions || triggerWriter.NamedExpressions.Count == 0) {
+          foreach (var triggerTerms in triggerWriter.Candidates) {
+            var trigger = new List<Expression>();
+            foreach (var triggerTerm in triggerTerms.Terms) {
+              trigger.Add(triggerTerm.Expr);
+            }
+
+            triggers.Add(trigger);
```

Comment:
(Reattaching this comment to the latest commit, I think this would read better as a Select, possibly wrapped in a SelectMany

Reply:
I changed the inner `foreach` to use `Select` (actually, I changed it to use `ConvertAll`, which is like `Select` but returns a list instead of an enumeration).